### PR TITLE
[Serialization] Simplify some code, tentatively fixing rdar://31812518.

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/Serialization/Serialization.cpp [hunk 1]
```diff
@@ -3845,7 +3845,7 @@ class YamlGroupInputParser {
   static llvm::StringMap<pFileNameToGroupNameMap> AllMaps;
 
   bool parseRoot(FileNameToGroupNameMap &Map, llvm::yaml::Node *Root,
-                 const llvm::Twine &ParentName) {
+                 std::string &ParentName) {
```
Comment (jrose-apple) [inline lib/Serialization/Serialization.cpp]: No need for `&` here. In fact, you can probably get away with StringRef.

**Review by jrose-apple — COMMENTED** [review]
Good call. Twine is just too tricky to use in non-immediate contexts.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Pull request merged**